guides/keras_cv/3d_object_detection_with_keras_cv.py

```diff
@@ -0,0 +1,367 @@
+sensors, and are generally structured as large batches of unsorted points
+captured at a single point in time.
+
+Before we get into the details of KerasCV's 3D object detection offerings, let's
+take a look at a pre-trained KerasCV 3D object detector. This preset has been
+trained on the entirety of the Waymo Open Dataset for 160 epochs, and achieves
+an average precision of 0.7 on the evaluation dataset for both vehicles and
+passengers.
+"""
+
+# TODO(ianstenbit): This doesn't work yet
+# pretrained_model = MultiHeadCenterPillar.from_preset("waymo_open_dataset_vehicles_and_pedestrians")
+# TODO(ianstenbit): Load up an eval frame and run predictions
+
+"""
+Great! Below we show a visualization of what some of these predictions look like
+alongside a rendering of the input point clouds.
+"""
+
+# TODO(ianstenbit): Include a link to a rendering of a visualization.
+
+"""
+Awesome! Now let's check out how to train this sort of model ourselves! First
+let's discuss how data is formatted for 3D object detection in KerasCV.
+In KerasCV, we adopt a data format where point clouds are represented as a
+dictionary with the following structure:
+
+```python
+point_cloud = {
+  "point_xyz": FloatTensor[batch_size, 3]
+  "point_features": FloatTensor[batch_size, num_features]
+  "point_mask": BooleanTensor[batch_size]
+}
+```
+
+The `point_xyz` field represents the XYZ coordinates of each point in the
+point cloud.
+
+The `point_features` field represents the LIDAR features of each point in the
+poin cloud. Typical features include range, intensity, and elongation.
+
+In KerasCV, 3D box targets for object detection are represented as vertical
+pillars rotated with respect to the Z axis. We encode each box as a list (or
+Tensor) of 7 floats: the X, Y, and Z coordinates of the box's center, the width,
+length, and height of the box, and the rotation of the box with respect to the
+Z axis. (This rotation is referrred to as `phi` and is always in radians).
+
+KerasCV's first 3D object detection model offering is a center-based model like
+the one proposed in https://arxiv.org/pdf/2006.11275.pdf.
+
+Let's get to 3D modelling!
+We'll start by loading up the Waymo Open Dataset. KerasCV provides a
+`waymo_open_dataset.load` function to load the Waymo Open Dataset into our
+data format.
+"""
+
+from keras_cv.datasets.waymo import load
+from keras_cv.datasets.waymo import build_tensors_for_augmentation
+
+# TODO(ianstenbit): Because we can't distribute WOD, we'll need to load wod_one_frame.tfrecord here
+data_shard = './training-data'
+dataset = load(data_shard)
+
+# By default, WOD point clouds are globally positioned, but for object detection
+# we want them with respect to the vehicle, so we transform them to the vehicle
+# frame of reference.
+dataset = dataset.map(transformer.transform_to_vehicle_frame)
+
+# Because number of points is dynamic, we pad them to make our inputs batchable.
+dataset = dataset.map(transformer.pad_or_trim_tensors)
```

Comment:
is there a ragged variant of the representation?  Does pad_or_trim_tensors take a HP for # of points in the cloud?